### PR TITLE
Conformance 1.2.1

### PR DESCRIPTION
**Pull request by fabricebrito**
Runs the conformance tests and and a mkdocs documentation

### CODE REVIEW

File: .github/workflows/docs.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  push:
+    branches:
+    - conformance-1.2.1
+    paths:
+    # Only rebuild website when docs have changed
+    - 'README.md'
+    - 'docs/**'
+    - 'mkdocs.yml'
+    - .github/**
+
```
Comment (emmanuelmathot) [inline .github/workflows/docs.yaml:12]: We probably want yo make a build only on release
```suggestion
  release:
    types: [created]
```

File: .github/workflows/docs.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+on:
+  push:
+    branches:
+    - conformance-1.2.1
+    paths:
+    # Only rebuild website when docs have changed
+    - 'README.md'
+    - 'docs/**'
+    - 'mkdocs.yml'
+    - .github/**
```
Comment (emmanuelmathot) [inline .github/workflows/docs.yaml:11]: probably better to build only on release
```suggestion
  release:
    types: [created]
```

Comment (fabricebrito) [reply .github/workflows/docs.yaml:11]: agreed

File: docs/result-v1-0.txt [hunk 3]
```diff
@@ -0,0 +1,725 @@
...
+
+Running: calrissian --max-ram 8G --max-cores 4 --default-container debian:stretch-slim --outdir=/output/tmpkoau27hd --quiet v1.0/v1.0/anon_enum_inside_array_inside_schemadef.cwl v1.0/v1.0/anon_enum_inside_array_inside_schemadef.yml
+Traceback (most recent call last):
+  File "/usr/local/bin/cwltest", line 8, in <module>
+    sys.exit(main())
+  File "/usr/local/lib/python3.10/site-packages/cwltest/main.py", line 236, in main
+    utils.generate_badges(args.badgedir, ntotal, npassed)
+  File "/usr/local/lib/python3.10/site-packages/cwltest/utils.py", line 132, in generate_badges
+    os.mkdir(badgedir)
+FileExistsError: [Errno 17] File exists: '/output/badges-1.0.2'
```
Comment (mr-c) [inline docs/result-v1-0.txt:725]: whoops :-)

Comment (fabricebrito) [reply docs/result-v1-0.txt:725]: will re-run. 
@mr-c with 1.2.1_proposed, I get `required` below 100%, something I will have to look into.
any clues?

**Comment by mr-c** [comment]
Looks like Calrissian is 100% for the required features of CWL v1.0, v1.1, and v1.2; so once those are posted I will move Calrission from "Partial implementations" to "In Production" at https://www.commonwl.org/implementations/#in-production ! (I will also link to the full conformance report)

Congratulations!

File: docs/result-v1-0.txt (hunk 3 above)
Comment (mr-c) [reply docs/result-v1-0.txt:725]: Here are the new mandatory tests in CWL v1.2.1:

For workflows: https://deploy-preview-115--cwl-v1-2-dev.netlify.app/workflow#New_Mandatory_Conformance_tests_for_v1.2.1 (just 1)
For CommandLineTools: https://deploy-preview-115--cwl-v1-2-dev.netlify.app/commandlinetool#New_Mandatory_Conformance_Tests_for_v1.2.1 (21 news mandatory tests)

Comment (mr-c) [reply docs/result-v1-0.txt:725]: Looks like the failing tests are
- params_broken_null [log](https://github.com/Duke-GCB/calrissian/pull/164/files#diff-83f52c3e68f7660d96d2f8f4fafcd728e7cfe76fc6b2a4d7737ba460fefa6b62R1933-R1958)
- length_for_non_array [log](https://github.com/Duke-GCB/calrissian/pull/164/files#diff-83f52c3e68f7660d96d2f8f4fafcd728e7cfe76fc6b2a4d7737ba460fefa6b62R1959-R1989)
- colon_in_paths [log of test timeout](https://github.com/Duke-GCB/calrissian/pull/164/files#diff-83f52c3e68f7660d96d2f8f4fafcd728e7cfe76fc6b2a4d7737ba460fefa6b62R1996-R2000)
- filename_with_hash_mark [log](https://github.com/Duke-GCB/calrissian/pull/164/files#diff-83f52c3e68f7660d96d2f8f4fafcd728e7cfe76fc6b2a4d7737ba460fefa6b62R2037-R2041)
- capture_files [log](https://github.com/Duke-GCB/calrissian/pull/164/files#diff-83f52c3e68f7660d96d2f8f4fafcd728e7cfe76fc6b2a4d7737ba460fefa6b62R2045-R2082)
- capture_dirs [log](https://github.com/Duke-GCB/calrissian/pull/164/files#diff-83f52c3e68f7660d96d2f8f4fafcd728e7cfe76fc6b2a4d7737ba460fefa6b62R2083-R2120)
- paramref_arguments_runtime: [log](https://github.com/Duke-GCB/calrissian/pull/164/files#diff-83f52c3e68f7660d96d2f8f4fafcd728e7cfe76fc6b2a4d7737ba460fefa6b62R2136-R2155)

**Review by emmanuelmathot — APPROVED** [review]

**Pull request merged**